webui/react/src/components/kit/SelectFilter.tsx

```diff
@@ -0,0 +1,149 @@
+import { Select } from 'antd';
+import type { DefaultOptionType, LabeledValue, RefSelectProps, SelectValue } from 'antd/es/select';
+import React, { forwardRef, useCallback, useMemo, useState } from 'react';
+
+import Label, { LabelTypes } from 'components/Label';
+import Icon from 'shared/components/Icon/Icon';
+
+import css from './SelectFilter.module.scss';
+
+const { OptGroup, Option } = Select;
+
+type Options = DefaultOptionType | DefaultOptionType[];
+export interface Props<T = SelectValue> {
+  allowClear?: boolean;
+  defaultValue?: T;
+  disableTags?: boolean;
+  disabled?: boolean;
+  enableSearchFilter?: boolean;
+  filterOption?: boolean | ((inputValue: string, option: LabeledValue | undefined) => boolean);
+  filterSort?: (a: LabeledValue, b: LabeledValue) => 1 | -1;
+  id?: string;
+  label?: string;
+  maxTagCount?: number | 'responsive';
+  maxTagPlaceholderValue?: string;
+  mode?: 'multiple' | 'tags';
+  onBlur?: () => void;
+  onChange?: (value: T, option: Options) => void;
+  onDeselect?: (selected: SelectValue, option: Options) => void;
+  onSearch?: (searchInput: string) => void;
+  onSelect?: (selected: SelectValue, option: Options) => void;
+  options?: LabeledValue[];
+  placeholder?: string;
+  ref?: React.Ref<RefSelectProps>;
+  value?: T;
+}
+
+export const ALL_VALUE = 'all';
+
+const countOptions = (children: React.ReactNode): number => {
+  if (!children) return 0;
+
+  let count = 0;
+  if (Array.isArray(children)) {
+    count += children.map((child) => countOptions(child)).reduce((acc, count) => acc + count, 0);
+  }
+
+  const childType = (children as React.ReactElement).type;
+  const childProps = (children as React.ReactElement).props;
+  const childList = (childProps as React.ReactPortal)?.children;
+  if (childType === Option) count++;
+  if (childType === OptGroup && childList) count += countOptions(childList);
+
+  return count;
+};
+
+const SelectFilter: React.FC<React.PropsWithChildren<Props>> = forwardRef(function SelectFilter(
```

Comment:
This is a nit but can we rename this to Dropdown or Select? It feels like the name is an artifact of when this was only used to select "filters".

Reply:
Otherwise I could change one of the other components to `AntdDropdown` or similar for the Select.